manifests/kpatch.pp

```diff
@@ -0,0 +1,26 @@
+# @summary Install kpatch
+#
+# @param enabled
+#   Install kpatch package and start service, noop if set to false (no clean-up)
+#
+# @param install_kpatch_patches
+#   Install the kpatch-patch package for the current kernel
+#
+# @example
+#   include profile_update_os::kpatch
+class profile_update_os::kpatch (
+  Boolean $enabled,
+  Boolean $install_kpatch_patches,
+) {
+  if $enabled {
+    ensure_packages( 'kpatch', { notify => Service['kpatch'] })
+
+    service { 'kpatch':
+      ensure => running,
+    }
+
+    if $install_kpatch_patches {
+      ensure_packages('kpatch-patch', { require => Service['kpatch'] })
```

Comment:
Would it make sense to have this ensure that the latest available is installed, instead of just any version? I can see going either way.

Ensuring latest would make sure that, if this was turned on and had already installed the "empty" version, that it would go on to update it if a new, more important version was released. If using snapshotted repos this could be controlled (although I realize we don't use those everywhere).

On the other hand, just ensuring installed would probably work out in most cases, as long as we keep this turned off by default and only set to true when there is an important version released?

Reply:
That's a good suggestion, I'll add that since having it verify it's the latest is worth it.

In my mind our normal default would be to leave `install_kpatch_patches` at false any only bump it to true when we had kpatches we need to install. But not everyone is going to think that way so having it always ensure the latest is an easy way to cover more bases.

Reply:
Makes sense, thanks!